test/common/common/safe_memcpy_test.cc

```diff
@@ -0,0 +1,48 @@
+#include <array>
+
+#include "common/common/safe_memcpy.h"
+#include "common/grpc/common.h"
+#include "common/http/headers.h"
+#include "common/http/message_impl.h"
+#include "common/http/utility.h"
+
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+using testing::ElementsAre;
+using testing::Eq;
+
+namespace Envoy {
+
+TEST(SafeMemcpyTest, CopyUint8) {
+  const uint8_t src[] = {0, 1, 1, 2, 3, 5, 8, 13};
+  uint64_t dst;
+  safeMemcpy(&dst, &src);
+  Eq(dst == 282583128934413);
```

Comment:
This comparison doesn't look right in two ways:

1. That pattern of using testing::Eq should probably be used in an EXPECT_THAT or similar. Usually for something like this, just use `EXPECT_EQ(dst, 282583128934413)`.
2. I think this might fail depending on the endian-ness of the machine where it is running. If your tests pass I think it's because this is not the way to use the `Eq` function.